src/goto-programs/show_goto_functions_json.cpp

```diff
@@ -0,0 +1,160 @@
+      json_functions.push_back(jsont()).make_object();
+    json_function["name"]=json_stringt(id2string(function_name));
+    json_function["isBodyAvailable"]=
+      jsont::json_boolean(function.body_available());
+    bool is_internal=(has_prefix(id2string(function_name), CPROVER_PREFIX) ||
+                      function_name==ID__start);
+    json_function["isInternal"]=jsont::json_boolean(is_internal);
+
+    if(function.body_available())
+    {
+      json_arrayt json_instruction_array=json_arrayt();
+
+      for(const goto_programt::instructiont &instruction :
+        function.body.instructions)
+      {
+        json_objectt instruction_entry=json_objectt();
+
+        instruction_entry["instructionId"]=
+          json_stringt(instruction.to_string());
+
+        if(instruction.code.source_location().is_not_nil())
+        {
+          instruction_entry["sourceLocation"]=
+            json(instruction.code.source_location());
+        }
+
+        std::ostringstream instruction_builder;
+        function.body.output_instruction(
+          ns, function_name, instruction_builder, instruction);
+
+        instruction_entry["instruction"]=
+          json_stringt(instruction_builder.str());
+
+        if(instruction.code.operands().size()>0)
+        {
+          json_arrayt operand_array;
+          for(const exprt &operand : instruction.code.operands())
+          {
+            json_objectt operand_object;
+            no_comments_irep_converter.convert_from_irep(
+              operand, operand_object);
+            operand_array.push_back(operand_object);
+          }
+          instruction_entry["operands"]=operand_array;
+        }
+
+        if(!instruction.guard.is_true())
+        {
+          json_objectt guard_object;
+          no_comments_irep_converter.convert_from_irep(
+            instruction.guard,
+            guard_object);
+
+          instruction_entry["guard"]=guard_object;
+        }
+
+        json_instruction_array.push_back(instruction_entry);
+      }
+
+      json_function["instructions"]=json_instruction_array;
+    }
+  }
+  json_objectt json_result;
+  json_result["functions"]=json_functions;
+  return json_result;
+}
+
+/*******************************************************************\
+
+Function: show_goto_functions_jsont::print_goto_functions
+
+  Inputs:
+   goto_functions - the goto functions that make up the program
+   out - the stream to write the object to
+   append - should a command and newline be appended to the stream
+            before writing the JSON object. Defaults to true
+
+ Outputs:
+
+ Purpose: Print the json object generated by
+          show_goto_functions_jsont::show_goto_functions to the provided
+          stream (e.g. std::cout)
+
+\*******************************************************************/
+
+void show_goto_functions_jsont::print_goto_functions(
+  const goto_functionst &goto_functions,
+  std::ostream &out,
+  bool append /*=true*/)
```

Comment:
You mean the `/*=true*/`? That was to indicate the default value, but I now realise I haven't seen that in the CBMC codebase. I'd suggest it is useful but I'm happy to remove if you disagree.